### DISCUSSION TITLE
Aqara H1 EU - Delay after click

### DISCUSSION THREAD

**Discussion by piotrGTX** [Koenkk/zigbee2mqtt#22252] [Devices]
I'm using several [WS-EUK02](https://www.zigbee2mqtt.io/devices/WS-EUK02.html#aqara-ws-euk02) devices.

Firmware Version: 01-12-2021

Zigbee2mqtt Docker Version: 1.35.3

Coordinator: Sonoff ZBDongle-P, zStack3x0 1.36.0

I bought these devices because they were among the few that supported **detach mode** and didn't require a **neutral** wire. 
In "control_relay" mode, the response is immediate. 

However, in "decoupled" mode, there is a significant delay of about 1.5-2 seconds after pressing the button before the message is received.

This delay is very annoying. Unaware users may think that the "system didn't react" and press the button again. I have to delay all actions with a 2-second drop to prevent this.....

I suspected it might be related to network performance. However, the message {"action": "off_right"} in "control_relay" mode appears in the queue immediately (which works fine).

Do all versions of this type of device have a similar issue? I haven't seen anyone mention this problem in reviews anywhere.

**Comment by julianfs** [comment]
I have the same switches & Sonoff-P (using the recommended 20230507 firmware) & I have a very small delay, way less than 1.5 - 2 seconds.

What's your experience turning on the lights from mqtt and/or z2m? Is that instant? Any grouping using HA helpers rather than using z2m groups?

**Comment by piotrGTX** [comment]
I tried to update the firmware to 20230507. Unfortunately, the whole network stopped working. Switches disconnected from the network and couldn't be paired again.

I managed to revert to the previous version 20221226. However, I didn't notice any improvement. Everything works as before.

Yes, actions like turning lights on/off from the application are immediate. This strange delay only applies to pressing the button in "detached" mode. 

I've prepared a video where you can see this delay clearly (~2 seconds). 
In the video, you can hear the relay clicking.

https://github.com/Koenkk/zigbee2mqtt/assets/10676069/60cc995d-fae3-4608-843b-93c27abaaac9

**Comment by julianfs** [comment]
I'm confused now, in decoupled mode there's no relay action required, I would expect that your switch is set permanently to 'On' and your switch event becomes something you react to in Home Assistant (or whatever you use) to turn toggle a smart bulb.

**Comment by piotrGTX** [comment]
I have two independent lamps in the corridor. For the purpose of the presentation, I set the left switch to detached mode and connected it to the second lamp. In Node-RED, I created a simple integration. If "action" == "single_left", set the relay to "ON". But this is irrelevant...

As you can see in the video, from the moment I press the button (single or double), it takes 2 seconds for the message to appear in the logs. That is, when I press the button, the second light turns on after 2 seconds.

Could you perform a similar test yourself ??? Set the "detach_mode". Check the logs. Click the button twice and see if the logs also appear after 2 seconds. You don't need to make any integrations.

**Comment by ghost** [comment]
Did u solve this? I am suspecting that the switch is waiting for the second click, so it must wait 1sec.

**Comment by llrp** [comment]
Any updates on this issue yet? Experiencing the same thing, it's super infuriating.

**Comment by llrp** [comment]
Anyone have any resolutions for this issue, custom firmware or anything?

**Comment by piotrGTX** [comment]
@llrp , @ghoz 

Small update. I bought the Aqara H2 and it solved the problem!

The new module has two modes (Multi-click feature):
- off - extremely low delay (almost immediate)
- on - low delay (~1s)

However, they’ve created a new problem... Now the internal relay has a delay.

For the H1, the relay changes state very quickly (almost immediately).
For the H2, the relay changes state slower (~1s).

I guess the Aqara H3 would solve both problems! 😄

But for me, H2 is much better than H1 👍
